### DISCUSSION TITLE
Compressed log

### DISCUSSION THREAD

**Discussion by acajic** [gabime/spdlog#3076] [Ideas]
This idea is similar to the one already suggested here https://github.com/gabime/spdlog/discussions/2064

Basically, if now I have a line like this:
`logger->info("test {}", var);`

I will find the following lines in my log file:
```
test 1
test 46
test 413
...

```

I would like to be able to pass additional argument what is the minimal time period between two logs of this type:
`logger->infoCompressed(5 * 60 /*5 minutes*/, "test {}", var);`

And in the log file I will find something like this:
`test 1 (+133 similar messages)`

and then after five minutes, we get another line like this:
`test 31 (+42 similar messages)`

What would be the best way to go about implementing this?

**Comment by tt4g** [comment]
It depends on whether the "similar messages" are log messages before or after formatting.
If it is a post-format message, this can only be implemented by sink with a formatter.

**Reply by acajic** [reply]
So, pre-format is "test {}" and post-format is "test 1"?
I need the pre-format.

**Reply by tt4g** [reply]
> So, pre-format is "test {}" and post-format is "test 1"?

Yes. And you can use dup_filter_sink, which has a very similar feature.
See dup_filter_sink section in Wiki: https://github.com/gabime/spdlog/wiki/4.-Sinks

You can also implement a custom sink by referring to dup_filter_sink.
See: https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink

**Reply by acajic** [reply]
Nice. Thank you.

> Each sink should be responsible for only single target (e.g file, console, db)

This may be a stupid question, but can multiple sinks point to the same target?

**Reply by tt4g** [reply]
Use dist_sink to proxy for multiple sinks.
By the way, dup_filter_sink inherits dist_sink.

**Comment by FilipWeisser** [comment]
I tried implementing this idea and ended up overloading and duplicating many functions. I had to add 2 parameters, "id" and "skip_duration", for all of info, warn, ..., log and sink_it functions. These 2 parameters are then used in a custom filter_similar_sink to keep count of the messages with the same id, and log them only if skip_duration has passed, with the (+x similar messages) message following it. eb2cb46f06d788af5e2649edda6a0426fa29861b

Is there a better way?
